Support ScalarDB Driver 3.17.1 (#3)
scalar-labs/cloudbeaver: README.md

<img src="https://github.com/dbeaver/cloudbeaver/wiki/images/cloudbeaver-logo.png" alt="CloudBeaver logo" align="right" width="250"/>

> # CloudBeaver Community Edition with ScalarDB

Reviewer: ```suggestion
# CloudBeaver Community Edition for ScalarDB
```
I'm not fully sure though. 
@josh-wong What do you think?

Reviewer 2: I'd suggest something like the following:

```suggestion
# CloudBeaver Community Edition with Support for ScalarDB
```

Or:

```suggestion
# CloudBeaver Community Edition with ScalarDB Integration
```

Author: I prefer Josh's first one. I'll apply it for now.

> CloudBeaver is a light modern web-application for the database management. Out-of-the-box CloudBeaver supports various SQL, NoSQL and BigData data sources. This repository provides a forked version that supports ScalarDB and free to use and open-source (licensed under [Apache 2](https://github.com/dbeaver/cloudbeaver/blob/devel/LICENSE) license). See the original [repository](https://github.com/dbeaver/cloudbeaver/) and [WIKI](https://github.com/dbeaver/cloudbeaver/wiki) for more details about CloudBeaver.

Author: Fixed in 6f8dec740.

<a><img src="./docs/images/connection-creation-demo.png" width="400"/></a>
<img src="./docs/images/data-editor-demo.png" width="400"/>
<img src="./docs/images/sql-editor-demo.png" width="400"/>
<img src="./docs/images/data-transfer-demo.png" width="400"/>

## Run in Docker

1. Prepare a client configuration file to connect ScalarDB Cluster. For details about the configuration, see [Client Configurations](https://scalardb.scalar-labs.com/docs/latest/scalardb-cluster/scalardb-cluster-configurations/#client-configurations).

1. Prepare a docker compose file and map the above configuration file to `/etc/scalardb.properties`. 

   ```yaml
   services:
     cloudbeaver:
       image: ghcr.io/scalar-labs/cloudbeaver:latest
       ports:
         - 8978:8978
       volumes:
         - ./workspace:/opt/cloudbeaver/workspace
         - ./scalardb.properties:/etc/scalardb.properties
   ```

1. Run the docker container

   ```console
   docker compose up -d
   ```

> ## How to use ScalarDB in CloudBeaver

Reviewer: ```suggestion
## How to access ScalarDB through CloudBeaver
```

- Once CloudBeaver is running, access http://localhost:8978/ and complete the setup based on [Administration](https://github.com/dbeaver/cloudbeaver/wiki/Administration).
- When creating a new connection setting, you will see ScalarDB in the database list and select it.
- Keep the file path in the Host field as is in the connection setting, and only change the Connection name as desired.
- If prompted for credentials when establishing the connection, no input is required unless Authentication is enabled (even when Authentication is enabled, no input is required if username/password are specified in the configuration file).

## Known Limitations

> - In SQL, [BEGIN](https://scalardb.scalar-labs.com/docs/latest/scalardb-sql/grammar#begin) is not supported, so use [START TRANSACTION](https://scalardb.scalar-labs.com/docs/latest/scalardb-sql/grammar#start-transaction) instead.

Reviewer: Can you clarify this?
`SQL` means here is the standard ANSI SQL?

Author: Sorry, I should have been clearer. To be precise, there is an issue in CloudBeaver's SQL editor where `BEGIN` is not recognized correctly when executing multiple lines of SQL as a script. That's why I intended to encourage the use of `START TRANSACTION` instead. Since starting a transaction with `BEGIN` is a DBMS-specific extension (e.g., [PostgreSQL](https://www.postgresql.org/docs/18/sql-begin.html)), CloudBeaver may not support it in the first place. However, since it works when executing line by line, it might simply be a bug. Anyway, how about changing like the following?

```suggestion
- Due to a limitation in how CloudBeaver's SQL editor parses scripts, [BEGIN](https://scalardb.scalar-labs.com/docs/latest/scalardb-sql/grammar#begin) may not work as expected when executing multiple statements. Use [START TRANSACTION](https://scalardb.scalar-labs.com/docs/latest/scalardb-sql/grammar#start-transaction instead), which is the SQL-standard command, instead.
```

Author: Revised in e59d54258.

## Run in Docker

1. Prepare a client configuration file to connect ScalarDB Cluster. For details about the configuration, see [Client Configurations](https://scalardb.scalar-labs.com/docs/latest/scalardb-cluster/scalardb-cluster-configurations/#client-configurations).

1. Prepare a docker compose file and map the above configuration file to `/etc/scalardb.properties`. 

   ```yaml
   services:
     cloudbeaver:
       image: ghcr.io/scalar-labs/cloudbeaver:latest
       ports:
         - 8978:8978
       volumes:
         - ./workspace:/opt/cloudbeaver/workspace
         - ./scalardb.properties:/etc/scalardb.properties
   ```

1. Run the docker container

   ```console
   docker compose up -d
   ```

## How to access ScalarDB through CloudBeaver

- Once CloudBeaver is running, access http://localhost:8978/ and complete the setup based on [Administration](https://github.com/dbeaver/cloudbeaver/wiki/Administration).
- When creating a new connection setting, you will see ScalarDB in the database list and select it.
- Keep the file path in the Host field as is in the connection setting, and only change the Connection name as desired.

> - If prompted for credentials when establishing the connection, no input is required unless Authentication is enabled (even when Authentication is enabled, no input is required if username/password are specified in the configuration file).

Reviewer: ```suggestion
- If prompted for credentials when establishing the connection, input a username and a password of a ScalarDB user. You can leave them empty if authentication is disabled or they are specified in the configuration file.
```
I think we should say something in case the authentication is enabled.

> ## Known Limitations

Reviewer: Should we note that the cross-partition scan must be enabled to execute the view?
Even if it's enabled by default, we don't recommend it for non-RDBMSs, so I want to note it somewhere.

Author: Added in 32a738d. Also, I added one more important limitation in the data editor because I think it takes a long time to fix it. For the `DATE` and `TIME` limitation, I will remove it in the next PR that introduces the ScalarDB custom plugin to fix it.
